lib/ansible/cli/adhoc.py

```diff
@@ -76,13 +76,23 @@ def parse(self):
 
     def _play_ds(self, pattern, async_val, poll):
         check_raw = self.options.module_name in ('command', 'win_command', 'shell', 'win_shell', 'script', 'raw')
+
+        mytask = {'action': {'module': self.options.module_name, 'args': parse_kv(self.options.module_args, check_raw=check_raw)}}
+
+        # avoid adding to tasks that don't support it, unless set, then give user an error
+        if self.options.module_name not in ('include_role', 'include_tasks'):
+            mytask['async_val'] = async_val
+            mytask['poll'] = poll
+        elif async_val:
+            mytask['async_val'] = async_val
+        elif poll:
```

Comment:
I don't think we want 2 `elif` statements here.  Maybe just `elif any((async_val, poll)):`